libvcx/src/connection.rs

```diff
@@ -618,4 +634,47 @@ pub mod tests {
         let empty = download_messages_noauth(Some(vec![unknown_did]), None, None).unwrap();
         assert_eq!(empty.len(), 0);
     }
+
+    #[cfg(feature = "agency_pool_tests")]
+    #[test]
+    fn test_download_messages() {
+        let _setup = SetupLibraryAgencyV2::init();
+        let consumer1 = create_consumer_config();
+        let consumer2 = create_consumer_config();
+
+        let (consumer1_to_institution, institution_to_consumer1) = create_connected_connections(Some(consumer1), None);
+        let (consumer2_to_institution, institution_to_consumer2) = create_connected_connections(Some(consumer2), None);
+
+        let consumer1_pwdid = get_their_pw_did(consumer1_to_institution).unwrap();
+        let consumer2_pwdid = get_their_pw_did(consumer2_to_institution).unwrap();
+
+        ::utils::devsetup::set_consumer(Some(consumer1));
+        send_generic_message(consumer1_to_institution, "Hello Institution from consumer1").unwrap();
+        ::utils::devsetup::set_consumer(Some(consumer2));
+        send_generic_message(consumer2_to_institution, "Hello Institution from consumer2").unwrap();
+
+        ::utils::devsetup::set_institution(None);
+        let all_msgs = download_messages([institution_to_consumer1, institution_to_consumer2].to_vec(), None, None).unwrap();
+        assert_eq!(all_msgs.len(), 2);
+        assert_eq!(all_msgs[0].msgs.len(), 2);
+        assert_eq!(all_msgs[1].msgs.len(), 2);
+
+        let consumer1_msgs = download_messages([institution_to_consumer1].to_vec(), None, None).unwrap();
+        assert_eq!(consumer1_msgs.len(), 1);
+        assert_eq!(consumer1_msgs[0].msgs.len(), 2);
+        assert_eq!(consumer1_msgs[0].pairwise_did, consumer1_pwdid);
+
+        let consumer2_msgs = download_messages([institution_to_consumer2].to_vec(), None, None).unwrap();
+        assert_eq!(consumer2_msgs.len(), 1);
+        assert_eq!(consumer2_msgs[0].msgs.len(), 2);
+        assert_eq!(consumer2_msgs[0].pairwise_did, consumer2_pwdid);
+
+        let consumer1_reviewed_msgs = download_messages([institution_to_consumer1].to_vec(), Some(vec![MessageStatusCode::Received]), None).unwrap();
```

Comment:
let's also check for payload, doesn't have to be per every `download_messages` result, but at least for
`consumer1_received_msgs` and `consumer1_reviewed_msgs `